wandb/sdk/artifacts/artifact.py

```diff
@@ -238,24 +238,33 @@ def _from_name(
         name: str,
         client: RetryingClient,
         organization: str = "",
+        enable_tracking: bool = False,
     ) -> Artifact:
+        server_supports_enabling_artifact_usage_tracking = (
+            InternalApi().server_project_type_introspection()
+        )
+        enable_tracking_arg = (
+            ", enableTracking: $enableTracking"
+            if server_supports_enabling_artifact_usage_tracking
+            else ""
+        )
         query = gql(
             """
             query ArtifactByName(
                 $entityName: String!,
                 $projectName: String!,
-                $name: String!
-            ) {
-                project(name: $projectName, entityName: $entityName) {
-                    artifact(name: $name) {
+                $name: String!,
+                $enableTracking: Boolean
+            ) {{
+                project(name: $projectName, entityName: $entityName) {{
+                    artifact(name: $name{enable_tracking_arg}) {{
```

Comment:
Would love to know if there's a cleaner way to do this 😅

Reply:
this is minor/optional, but f-strings might clean this up a bit.  I.e editing the following snippet:

(before)
https://github.com/wandb/wandb/blob/ad254f0099b1daf1201eb503f2c368304f303ad6/wandb/sdk/artifacts/artifact.py#L246-L275

...to something like...

(after)
```python3
        query_vars = ["$entityName: String!", "$projectName: String!", "$name: String!"]
        query_args = ["name: $name"]        
        if server_supports_enabling_artifact_usage_tracking:
            query_vars = [*query_vars, "$enableTracking: Boolean"]
            query_args = [*query_args, "enableTracking: $enableTracking"]

        vars_str = ", ".join(query_vars)
        args_str = ", ".join(query_args)

        query = gql(
            f"""
            query ArtifactByName({vars_str}) {{
                project(name: $projectName, entityName: $entityName) {{
                    artifact({args_str}) {{
                        ...ArtifactFragment
                    }}
                }}
            }}
            {cls._get_gql_artifact_fragment()}
            """
        )
```

(GH doesn't allow an inline suggestion over the full block 😕)

Reply:
side note: `gql` does seem to allow [programmatically constructing GQL queries via the `gql.dsl` module](https://gql.readthedocs.io/en/latest/advanced/dsl_module.html#compose-queries-dynamically).  But I remember looking into this briefly for something else, and ran into issues/complications -- e.g. among other things, I think the underlying `Client` needs to be reconfigured as well.

(Also, internally we depend on an older version of `gql`, so the examples in the linked docs for `gql==3.*` don't quite work out-of-the-box)

Reply:
Oh wow, this module seems pretty powerful. Would have helped to have used it in a bunch of places in our internal API too!